src/components/tBTC/BridgeActivity.tsx

```diff
@@ -85,14 +87,14 @@ export const BridgeAcivityHeader: FC<StackProps> = (props) => {
 }
 
 export const BridgeActivityData: FC<ListProps> = (props) => {
-  const { data, isBridgeHistoryEmpty, isFetching, emptyState } =
+  const { data, isFetching, emptyState, isWalletConnected } =
     useBridgeActivityContext()
 
   return isFetching ? (
     <BridgeActivityLoadingState />
   ) : (
     <List spacing="1" mt="2" {...props}>
-      {isBridgeHistoryEmpty ? emptyState : data.map(renderActivityItem)}
+      {!!data && isWalletConnected ? data.map(renderActivityItem) : emptyState}
```

Comment:
Doesn't `!!` return true for and empty array? I think it does.

In that case it would be best to return `isBridgeHistoryEmpty` to the bridge activity context and use it instead of `!!data`.

Reply:
Ref commit: cfb65f4d